lnprototest/backend/bitcoind.py

```diff
@@ -144,11 +159,11 @@ def start(self) -> None:
         self.rpc.generatetoaddress(100, self.rpc.getnewaddress())
 
     def stop(self) -> None:
+        self.rpc.stop()
         self.proc.kill()
         shutil.rmtree(os.path.join(self.bitcoin_dir, "regtest"))
+        self.rpc = None
 
     def restart(self) -> None:
-        # Only restart if we have to.
-        if self.rpc.getblockcount() != 101 or self.rpc.getrawmempool() != []:
-            self.stop()
-            self.start()
+        self.stop()
+        self.start()
```

Comment:
You should restore this optimization!  But...

restart() is never actually called by the way..  You need to fix up clightning not to call stop and start on bitcoind when it gets reset.

Reply:
Yes, good catch, I used your trick with a bitcoind flag